src/module/ovs/bpf/kernel_exec_actions.bpf.c

```diff
@@ -15,8 +15,11 @@ DEFINE_HOOK(F_AND, RETIS_F_PACKET_PASS,
 	queue_id = queue_id_gen_skb(skb);
 
 	if (bpf_map_lookup_elem(&flow_exec_tracking, &queue_id)) {
-		/* Indicate this flow execution is the result of a userpace
-		 * command */
+		/* Indicate this flow execution is the result of a userspace
+		 * command and store the current queue_id so that further
+		 * actions will use the same one regardless of packet
+		 * modifications. */
+		ectx.queue_id = queue_id;
```

Comment:
If queue_id is not being tracked the `flow_exec` hook won't add it to the `flow_exec_tracking` map and the lookup will fail, not executing this code.

Reply:
> Right, let me rephrase. If the queue id is not tracked, the queue id won't be stored in `ectx`. Later (?) in `ovs_do_execute_action` we have:
> 
> ```
> track->queue_id = ectx->queue_id;
> ```
> 
> Before this commit the queue id was always recomputed there and couldn't be 0. Now it might be (?).

We only have `track->queue_id = ectx->queue_id` if `ectx->command`, which is set to true only when queue_id was being tracked (i.e: this comes from a tracked flow command)